qiskit/visualization/bloch.py

```diff
@@ -396,7 +396,7 @@ def add_annotation(self, state_or_vector, text, **kwargs):
         if isinstance(state_or_vector, (list, np.ndarray, tuple)) and len(state_or_vector) == 3:
             vec = state_or_vector
         else:
-            raise Exception("Position needs to be specified by a qubit " + "state or a 3D vector.")
+            raise RuntimeError("Position needs to be specified by a qubit state or a 3D vector.")
```

Comment:
This is maybe a `TypeError` because we'd have been given a completely incorrect type object.